### DISCUSSION TITLE
[Feature Request] Add Follow Buttons for Fediverse Services like Mastodon, PeerTube etc

### DISCUSSION THREAD

**Discussion by Thatoo** [GetPublii/Publii#1500] [Ideas]
It would be nice to haven in the "simple" theme (default theme), the ability to add follow buttons for

    Mastodon
    PeerTube
    FunkWhale
    Freindica
    PixelFed
    Mobilizon

and other services that are using the AcitivityPup protokoll.

It has been requested in : 
https://github.com/GetPublii/Publii/issues/1192
https://github.com/GetPublii/Publii/issues/1176
https://github.com/GetPublii/Publii/issues/1053

As the forum is moving to here, I open the discussion here now.

**Comment by ghost** [comment]
I bump up this suggestion. Is there perhaps a plan, to introduce in the add-on the possibility to plug in custom platforms and icons?

**Comment by bobmitro** [comment]
I have no plans to add more to the theme. You can add it yourself or use our [Follow Buttons](https://marketplace.getpublii.com/plugins/follow-buttons/) plugin, which is available in the Publii marketplace. 
There is only Mastodon from the @Thatoo list, so I will consider adding more in the next release.

If you want to add more platforms/services, please list them here, just make sure they aren't too niche.

**Comment by ghost** [comment]
I have Follow Buttons already. 

> just make sure they aren't too niche.

That's why could a great idea to possibility to make custom ones inside plugin.

**Comment by bobmitro** [comment]
The plugin provides many configuration options, such as icons that must be adequately prepared, allowing loading them from a single SVG sprite file or embedding them inline in the code. Adding a custom platform would involve a lot of work, and later, there would still be 100 questions about why this option or that icon isn't working correctly. So, I don't expect such an option in the plugin.

**Comment by ghost** [comment]
I understand. Pity. I unfortunately, in exception to the Fediverse platforms mentioned above (I would add Lemmy or GoToSocial), would have suggestions for niche platforms. Besides, Fediverse itself is also niche. I would like to add some podcast platforms especially those open-sourced and without tracking, because for the moment there are only the mainstream Apple Podcast and Spotify, which are de facto not good for podcast interoperability (Spotify provides its own files extracted from RSS, it provides neither RSS nor a link to its own site, Apple Podcast is a closed database). 

I would like, for example, a link to PodcastIndex. A competitor to Apple's open sourced database for podcasts.

**Reply by bobmitro** [reply]
Please list them one below the other in a code box:

```
item1
item2
item3
```

**Comment by ghost** [comment]
podcast:

```
https://podcastindex.org
https://podverse.fm
https://antennapod.org
https://podcastaddict.com
https://castopod.org/
RSS
```

fediverse:
```
Lemmy
GoToSocial
Castopod
```

Blog RSS
`RSS`

ps.

However, consider maybe links to custom platforms without custom icons. The way I imagine it is that the icons are circles of different colors to choose from. The most important thing, however, is that I can create custom name, which should not be a problem. Currently now I also give a link to custom platform, but unfortunately the name cannot be changed.

**Comment by bobmitro** [comment]
As promised, we've just released a new update for the plugin, version 1.5.0, including several new platforms:
- Antennapod
- Castopod
- Freindica
- FunkWhale
- Lemmy
- PeerTube
- Pixelfed
- Podcast Addict
- Podcast Index
- Podverse
- RSS
<img width="428" alt="screenshot 2024-08-01 at 12 18 18" src="https://github.com/user-attachments/assets/eaa3564b-94ee-492c-a828-a24cfe4374fc">

Download the [followButtons](https://marketplace.getpublii.com/plugins/follow-buttons/) plugin!

**Comment by ghost** [comment]
Thank You. Great.

**Comment by trioptimum** [comment]
Any chance of a Follow Buttons update to add Bluesky, please? As the service that many Twitter refugees seem to be landing on (almost 25 million registered users, up from 9 million in September), it's a noticeable omission. Thanks.

**Reply by bobmitro** [reply]
Added :) Just update your installation to the new [1.5.4 plugin version ](https://marketplace.getpublii.com/plugins/follow-buttons/).

**Reply by trioptimum** [reply]
Confirming it works, thanks for the fast turnaround!
